pkg/chains/formats/slsa/v2alpha2/internal/build_definitions/resolved_dependencies.go

```diff
@@ -41,13 +42,52 @@ const (
 	pipelineResourceName = "pipelineResource"
 )
 
+// used to toggle the fields in resolvedDependencies. see AddTektonTaskDescriptor
+// and AddSLSATaskDescriptor
+type addTaskDescriptorContent func(*v1beta1.TaskRun) (v1.ResourceDescriptor, error)
+
+// the more verbose resolved dependency content. this adds the name, uri, digest
+// and content if possible.
+func AddTektonTaskDescriptor(tr *v1beta1.TaskRun) (v1.ResourceDescriptor, error) {
+	rd := v1.ResourceDescriptor{}
+	storedTr, err := json.Marshal(tr)
+	if err != nil {
+		return rd, err
+	}
+	logger := logging.FromContext(context.TODO())
+	logger.Infof("logging taskRun %v", tr)
+	// add remote task configsource information in materials
+	if tr.Status.Provenance != nil && tr.Status.Provenance.RefSource != nil {
+		rd.Name = pipelineTaskConfigName
+		rd.URI = tr.Status.Provenance.RefSource.URI
+		rd.Digest = tr.Status.Provenance.RefSource.Digest
+		rd.Content = storedTr
+	} else {
+		rd.Name = pipelineTaskConfigName
+		rd.Content = storedTr
+	}
+	return rd, nil
+}
+
+// resolved dependency content for the more generic slsa verifiers. just logs
+// the name, uri and digest.
+func AddSLSATaskDescriptor(tr *v1beta1.TaskRun) (v1.ResourceDescriptor, error) {
+	rd := v1.ResourceDescriptor{}
+	if tr.Status.Provenance != nil && tr.Status.Provenance.RefSource != nil {
+		rd.Name = pipelineTaskConfigName
+		rd.URI = tr.Status.Provenance.RefSource.URI
+		rd.Digest = tr.Status.Provenance.RefSource.Digest
+	}
+	return rd, nil
+}
+
 // TaskRun constructs `predicate.resolvedDependencies` section by collecting all the artifacts that influence a taskrun such as source code repo and step&sidecar base images.
-func TaskRun(ctx context.Context, tro *objects.TaskRunObject) ([]v1.ResourceDescriptor, error) {
+func (t TaskBuildType) ResolvedDependencies(ctx context.Context) ([]v1.ResourceDescriptor, error) {
```

Comment:
I find it confusing to attach a method to a type defined in a different file. Do we need to make this change?